integration/benchmarks/rax/run_benchmark.py

```
@@ -0,0 +1,730 @@
+    parser.add_argument("--baseline-csv", default=None, help="Path to pre-captured baseline benchmark CSV to reuse historical results without re-running baseline benchmarks")
+    parser.add_argument("--threads", nargs="+", type=int, default=[1, 4, 8, 16], help="Thread counts to test")
+    parser.add_argument("--setups", nargs="+", choices=["text_only", "text_tag_mixed", "text_tag_churn_prefix", "all"], default=["all"], help="Benchmark setups to run")
+    parser.add_argument("--append", action="store_true", help="Append results to existing CSV instead of overwriting")
+    args = parser.parse_args()
+
+    if not os.path.exists(args.server):
+        raise FileNotFoundError(f"valkey-server not found at {args.server}")
+    if not os.path.exists(args.module):
+        print(f"[INFO] libsearch.so not found at '{args.module}'. Building automatically...")
+        build_script = os.path.join(PROJECT_ROOT, "build.sh")
+        subprocess.run([build_script], check=True, cwd=PROJECT_ROOT)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)